src/qasm3/Importer.cpp

```diff
@@ -672,10 +702,32 @@ std::unique_ptr<qc::Operation> Importer::evaluateGateCall(
       allQubits.emplace(qubit);
     }
 
-    // first we apply the operation
-    auto nestedOp = applyQuantumOperation(gate, targetBits, controlBits,
+    // first we apply the operation, potentially multiple times
+    std::unique_ptr<qc::Operation> nestedOp = nullptr;
+    for (int64_t p = 0; p < powerExponent; ++p) {
+      auto single = applyQuantumOperation(gate, targetBits, controlBits,
                                           evaluatedParameters, invertOperation,
                                           gateCallStatement->debugInfo);
+      if (single == nullptr) {
+        nestedOp = nullptr;
+        break;
+      }
+      if (powerExponent == 1) {
+        nestedOp = std::move(single);
+        break;
+      }
```

Comment:
I believe this check is redundant as this is essentially what will happen automatically if `powerExponent == 1`.
```suggestion
```